lib/fs/filesystem.go

```diff
@@ -273,9 +288,9 @@ func Canonicalize(file string) (string, error) {
 // It ensures proper wrapping order, which right now means:
 // `logFilesystem` needs to be the outermost wrapper for caller lookup.
 func wrapFilesystem(fs Filesystem, wrapFn func(Filesystem) Filesystem) Filesystem {
-	logFs, ok := fs.(*logFilesystem)
+	logFs, ok := UnwrapFilesystem(fs, FilesystemWrapperTypeLog)
```

Comment:
I'd keep the direct type assertion: This functions is about the order of wrapping, while `UnwrapFilesystem` doesn't care about the order and just gets the wrapper from any "depth". And as the comment above explains, the logging wrapper needs to be the outermost at all times.